src/Microsoft.AspNet.Mvc.Core/ModelBinding/Metadata/DefaultModelMetadata.cs

```diff
@@ -221,6 +227,48 @@ public override string EditFormatString
         }
 
         /// <inheritdoc />
+        public override ModelMetadata ElementMetadata
+        {
+            get
+            {
+                if (!_haveCalculatedElementMetadata)
+                {
+                    // Short-circuit checks below. As in IsCollectionType, do not consider string a collection.
+                    _haveCalculatedElementMetadata = ModelType == typeof(string);
+                }
+
+                if (!_haveCalculatedElementMetadata)
```

Comment:
OK, I'll move the `string` check after `_haveCalculatedElementMetadata` is set to `true` and `return` in that case.